fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -318,6 +323,71 @@ private void updateDimensionRow(
     }
 
     /**
+     * Replaces Lucene Index with a new index.
+     *
+     * @param newLuceneIndexPath  The path of the new index
+     */
+    public void replaceIndex(String newLuceneIndexPath) {
+        Directory oldLuceneDirectory = luceneDirectory;
+
+        lock.writeLock().lock();
+        try {
+            // move the old dir to a different location
+            try {
+                Files.move(
+                        new File(luceneIndexPath).getAbsoluteFile().toPath(),
+                        new File(renameToTemp(luceneIndexPath)).getAbsoluteFile().toPath(),
+                        REPLACE_EXISTING
+                );
+            } catch (IOException e) {
+                String message = ErrorMessageFormat.CANNOT_MOVE_INDEX_DIR.format(luceneIndexPath);
+                LOG.error(message, e);
+            }
+
+            // open a new lucene index dir
+            try {
+                luceneDirectory = new MMapDirectory(Paths.get(newLuceneIndexPath));
+            } catch (IOException e) {
+                String message = ErrorMessageFormat.CANNOT_CREATE_INDEX_DIR.format(newLuceneIndexPath);
+                LOG.error(message, e);
+            }
+
+            luceneIndexPath = newLuceneIndexPath; // have lucene point to the new lucene index dir
+
+            reopenIndexSearcher(true);
+        } finally {
+            lock.writeLock().unlock();
+        }
+
+        // delete old dir
+        try (IndexWriter indexWriter = new IndexWriter(
+                oldLuceneDirectory,
+                new IndexWriterConfig(LUCENE_ANALYZER).setRAMBufferSizeMB(BUFFER_SIZE)
+        )) {
+            indexWriter.close();
```

Comment:
The reason I use `IndexWriter.close()` comes from 

https://stackoverflow.com/questions/12499108/how-to-delete-lucene-directory-along-with-different-segment-files

https://stackoverflow.com/questions/21901499/why-i-cannot-delete-index-directories-in-lucene-after-i-make-the-searcher-null